frontend/src/api/endpoints.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * API endpoint functions using generated types
+ */
+
+import type {
+  BodyLoginTokenPost,
+  Token,
+  User,
+  ChatRequest,
+} from '../types/generated';
+import { apiClient } from './client';
+import { readStream, type StreamOptions } from './streaming';
+
+/**
+ * JWT token login
+ */
+export async function login(
+  username: string,
+  password: string,
+): Promise<Token> {
+  const formData: BodyLoginTokenPost = {
+    username,
+    password,
+    grant_type: 'password',
+  };
+
+  const token = await apiClient.postForm<Token>('/token', {
+    username: formData.username,
+    password: formData.password,
+    grant_type: formData.grant_type || '',
+  });
+
+  apiClient.setToken(token.access_token);
+
+  return token;
+}
+
+/**
+ * Logout
+ */
+export function logout(): void {
+  apiClient.clearToken();
+}
+
+/**
+ * Get current user
+ */
+export async function getCurrentUser(): Promise<User> {
+  return apiClient.get<User>('/me');
+}
+
+/**
+ * Send a chat message and receive streaming responses
+ *
+ * @param query - The user's question/message
+ * @param options - Streaming options (onMessage, onError, onComplete)
+ * @param useSSE - Whether to use Server-Sent Events (true) or NDJSON (false). Default: true
+ */
+export async function sendChatMessage(
+  query: string,
+  options: StreamOptions,
+): Promise<void> {
+  const requestBody: ChatRequest = { query };
+
+  const headers: Record<string, string> = {
+    'Content-Type': 'application/json',
+    Accept: 'text/event-stream',
```

Comment:
Just a note, since we're using event stream, we could use [events](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events#event_stream_format) to route information down the road. I don't think it's required yet.